### PR TITLE
feat: add solution for day 17

### PR DESCRIPTION
**Pull request by hunghoang3110**
## Summary by CodeRabbit

- **New Features**
	- Introduced new test specifications for user registration and product addition functionalities.
	- Added a solution for a coding challenge that processes various input scenarios with error handling.

- **Bug Fixes**
	- Enhanced validation checks for user inputs during registration and product addition.

- **Tests**
	- Implemented end-to-end tests using Playwright for verifying user registration and shopping cart functionality.

### CODE REVIEW

File: q
Comment (minhphong306) [inline q:1]: Please remove un-related files

File: daily-challenges/2024-09/17/hunghoang3110.spec.ts
Comment (minhphong306) [inline daily-challenges/2024-09/17/hunghoang3110.spec.ts:1]: Please split the challenge into smaller ones. 
One PR should only contain one day's challenge.

**Review by minhphong306 — CHANGES_REQUESTED** [review]
Please take a look at my comments.

File: q
Comment (hunghoang3110) [reply q:1]: I got it

File: daily-challenges/2024-09/17/hunghoang3110.spec.ts
Comment (hunghoang3110) [reply daily-challenges/2024-09/17/hunghoang3110.spec.ts:1]: Copied that

**Review by minhphong306 — APPROVED** [review]
LGTM

**Pull request merged**